scimath/interpolate/py2to3.h

```diff
@@ -1,256 +0,0 @@
-#endif
-
-PyObject *Py2to3_GetAttrDictValue(PyDictObject * dict, PyObject *name, PyObject *bad_attr) {
-
-    long hash;
-    PyObject *nname;
-    PyObject *value;
-    if ( PyString_CheckExact( name ) ) {
-        if ( (hash = ((PyStringObject *) name)->ob_shash) == -1 )
-           hash = PyObject_Hash( name );
-        return (dict->ma_lookup)( dict, name, hash )->me_value;
-    }
-
-    nname = Py2to3_NormaliseAttrName(name);
-    if( nname == NULL ){
-        PyErr_Clear();
-        return bad_attr;
-    }
-
-    hash = PyObject_Hash( nname );
-    if( hash == -1 ){
-        Py2to3_FinishNormaliseAttrName(name,nname);
-        PyErr_Clear();
-        return NULL;
-    }
-    value = (dict->ma_lookup)( dict, nname, hash )->me_value;
-    Py2to3_FinishNormaliseAttrName(name,nname);
-    return value;
-}
-
-#else // #if PY_MAJOR_VERSION < 3
-
-#define Py2to3_AttrNameCheck(name) (PyUnicode_Check(name))
-
-PyObject *Py2to3_GetAttrDictValue(PyDictObject * dict, PyObject *name, PyObject *bad_attr) {
-#if !defined(Py_LIMITED_API) && (PY_MAJOR_VERSION < 3 || PY_MINOR_VERSION < 3) && 0 // this does not work as intended! don't use it!
-    Py_hash_t hash;
-    if( PyUnicode_CheckExact( name ) ) {
-#if PY_MINOR_VERSION < 3
-        hash = ((PyUnicodeObject *) dict)->hash;
-#else
-        // currently not usable, as dict->ma_lookup does not exist either
-        hash = ((PyUnicodeObject *) dict)->_base._base.hash;
-#endif
-        if( hash == -1)
-           hash = PyObject_Hash( name );
-        return (dict->ma_lookup)( dict, name, hash )->me_value;
-    }
-#endif // #ifndef Py_LIMITED_API
-    if( !PyUnicode_Check(name) )
-        return bad_attr;
-
-    return PyDict_GetItem((PyObject *)dict, name);
-}
-
-#endif // #if PY_MAJOR_VERSION < 3
-
-
-/*
-*/
-
-double Py2to3_PyNum_AsDouble(PyObject *value) {
-#if PY_MAJOR_VERSION < 3
-    if ( PyInt_Check( value ) ) {
-        return (double) PyInt_AS_LONG( value );
-    } else if( !PyLong_Check(value) ){
-#else
-    if ( !PyLong_Check( value ) ) {
-#endif  // #if PY_MAJOR_VERSION < 3
-        PyErr_SetNone( PyExc_TypeError );
-        return -1.;
-    }
-    return PyLong_AsDouble( value );
-}
-
-
-#ifndef PyVarObject_HEAD_INIT
-    #define PyVarObject_HEAD_INIT(type, size) \
-        PyObject_HEAD_INIT(type) size,
-#endif
-
-
-
-#if PY_MAJOR_VERSION >= 3
-  #define Py2to3_MOD_ERROR_VAL NULL
-  #define Py2to3_MOD_SUCCESS_VAL(val) val
-  #define Py2to3_MOD_INIT(name) PyMODINIT_FUNC PyInit_##name(void)
-  #define Py2to3_MOD_DEF(ob, name, doc, methods) \
-          static struct PyModuleDef moduledef = { \
-            PyModuleDef_HEAD_INIT, name, doc, -1, methods, }; \
-          ob = PyModule_Create(&moduledef);
-#else
-  #define Py2to3_MOD_ERROR_VAL
-  #define Py2to3_MOD_SUCCESS_VAL(val)
-  #define Py2to3_MOD_INIT(name) void init##name(void)
-  #define Py2to3_MOD_DEF(ob, name, doc, methods) \
-          ob = Py_InitModule3(name, methods, doc);
-#endif
```

Comment:
FTR, this is the only relevant bits of code in `py2to3.h` which are used in `_interpolate.cpp`